### ISSUE TITLE
continuous loader source and destination entity count does not match after shrink()

### ISSUE THREAD

**Issue by Siltnamis** [skypjack/entt#588]
Hi, i'm playing around with archives and continuous_loader and encountered a strange issue where entity count for source and destination registries does not match. Here is a small example of couple components and systems.

```
#include <random>
#include <chrono>
#include <entt.hpp>
#include <cereal/archives/json.hpp>
#include <thread>
#include <assert.h>
#include <stdio.h>

struct position {float x; float y;};
struct lifetime {uint64_t end_of_life; };
typedef entt::tag<"destroy"_hs> destroy_tag;

template<typename Archive>
void serialize(Archive &archive, position &pos) {
    archive(pos.x, pos.y);
}

template<typename Archive>
void serialize(Archive &archive, lifetime &life) {
    archive(life.end_of_life);
}

std::mt19937 rng;
float roll()
{
    return std::uniform_real_distribution<float>(0.0f, 1.0f)(rng);
}

uint64_t time_stamp(std::chrono::milliseconds dur)
{
    auto now = std::chrono::high_resolution_clock::now();
    return std::chrono::duration_cast<std::chrono::milliseconds>((now + dur).time_since_epoch()).count();
}

void spawn(entt::registry& reg)
{
    entt::entity e = reg.create();   
    reg.emplace<position>(e, position{roll(), roll()});
    reg.emplace<lifetime>(e, lifetime{time_stamp(std::chrono::milliseconds(100))});
}
void move(entt::registry& reg)
{
    reg.view<position>().each([](entt::entity e, position& pos){ pos.x += 0.1; });
}

void life_time(entt::registry& reg)
{
    reg.view<lifetime>().each([&reg](const entt::entity e, lifetime& life)
    {
    uint64_t now = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::high_resolution_clock::now().time_since_epoch()).count();
    if(now >= life.end_of_life)
        reg.emplace_or_replace<destroy_tag>(e);
    });
}

void nuke(entt::registry& reg)
{
    if(reg.alive() > 100000)
    {
        reg.each([&reg](entt::entity e)
        {
            reg.emplace_or_replace<destroy_tag>(e);
        });
    }
}

void destroy(entt::registry& reg)
{
    auto to_destroy = reg.view<destroy_tag>();
    reg.destroy(to_destroy.begin(), to_destroy.end());
};

int main()
{
    entt::registry main_registry;
    entt::registry destination;
    entt::continuous_loader loader{destination};
    while(1)
    {
        spawn(main_registry);
        move(main_registry);
        life_time(main_registry);
        nuke(main_registry);
        destroy(main_registry);
        std::stringstream storage;
        {
        cereal::JSONOutputArchive output(storage);
        entt::snapshot{main_registry}.entities(output)
            .component<position>(output);
        }
        cereal::JSONInputArchive input(storage);
        loader.entities(input).component<position>(input).orphans().shrink();

        if(main_registry.alive() != destination.alive()) // should this ever be true?
        {
            printf("%d\t%d\n", (int)main_registry.alive(), (int)destination.alive());
        }
        std::this_thread::sleep_for(std::chrono::milliseconds(5));
    }
    return 0;
}
```

If you run this code you will see different numbers being printed out.
Either I am doing something illegal or I'm not understanding archives/loaders correctly.

**Comment by skypjack** [comment]
Hi, can you tell me what version of EnTT you're using? Thanks.

**Comment by Siltnamis** [comment]
I just cloned current master

**Comment by Siltnamis** [comment]
Just tried with latest **released** version (v3.5.2) and this issue is not present. My fault for not using released version. But it may be a good idea to investigate why it behaves like this with current master.

**Comment by skypjack** [comment]
Indeed. Thanks for pointing this out. Please, do not close the issue. I'll look into it soon!
